nomenclature/codelist.py

```diff
@@ -96,6 +98,12 @@ def keys(self):
     def values(self):
         return self.mapping.values()
 
+    def validate_dataframe(self, df: IamDataFrame, dimension: str) -> bool:
```

Comment:
Suggest to rename because only the `data` attribute is validated, not `meta`.
```suggestion
    def validate_data(self, df: IamDataFrame, dimension: str) -> bool:
```